### PR TITLE
ramips: add support for Buffalo WSR-2533DHPLS/DHPL2

### PR DESCRIPTION
**Pull request by musashino205**
This patch series adds support for Buffalo WSR-2533DHPLS and WSR-2533DHPL2 using TRX and UBI on NAND flash, like the WSR series devices on mediatek/mt7622.

This PR is a successor of #14835.

### CODE REVIEW

**Comment by musashino205** [comment]
UPDATE:

- rebase from current main branch
- switch trx parser to parser_trx on Linux 6.6

**Comment by audunmg** [comment]
Finally got around to testing it, and it works great on my two WSR-2533DHPLS

Thank you!

**Comment by musashino205** [comment]
UPDATE:

- rebase from current main branch
- update network nodes for the current binding
- disable unused pcie2

**Comment by musashino205** [comment]
UPDATE:

- rebase from current main branch

File: target/linux/ramips/mt7621/base-files/etc/uci-defaults/09_fix_crc [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+fstype="$(/bin/mount | awk '($3 ~ /^\/$/) && ($5 !~ /rootfs/) { print $5 }')"
+[ "$fstype" = "tmpfs" ] && \
+	exit 0
+
+case "$(board_name)" in
+buffalo,wsr-2533dhpls)
+	trx_magic=0x44484C53
+	;;
+*)
+	exit 0
```
Comment (hauke) [inline target/linux/ramips/mt7621/base-files/etc/uci-defaults/09_fix_crc:16]: I do not like the exit in the default branch.
I would prefer if you put the line calculating `kernel_size` and calling `mtd` into a function which gets called for `buffalo,wsr-2533dhpls`.

Comment (musashino205) [reply target/linux/ramips/mt7621/base-files/etc/uci-defaults/09_fix_crc:16]: I see, I'll update this.

**Pull request merged**